src/httpserver.py

```diff
@@ -246,7 +246,8 @@ def do_POST(self):
                          'CONTENT_TYPE':self.headers['Content-Type'],
                          })
         else:
-            form = self.rfile.read()
+            content_length = int(self.headers['Content-Length']) if 'Content-Length' in self.headers else 0
```

Comment:
Do `content_length = int(self.headers.get('Content-Length', '0'))` here, it's shorter.